bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/ProgressCallback.java

```diff
@@ -74,4 +75,14 @@
      * Callback operation to indicate that the firmware update was canceled.
      */
     void canceled();
+    
+    /**
+     * Callback operation to indicate that the percentage progress of the firmware update has changed. 
+     * 
+     * @param progress the progress between 0 and 100
+     * @param silent if true EventSubscriber want get informed 
+     * 
+     * @throws IllegalArgumentException if given progress is < 0 or > 100
```

Comment:
We should think about another name of this operation... the other operations of this interface does not start with set... maybe next(int progress) or update(int progress)?